pkg/registry/ingress/strategy.go

```diff
@@ -114,3 +116,20 @@ func MatchIngress(label labels.Selector, field fields.Selector) generic.Matcher
 		},
 	}
 }
+
+type ingressStatusStrategy struct {
+	ingressStrategy
+}
+
+var StatusStrategy = ingressStatusStrategy{Strategy}
+
+func (ingressStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newIngress := obj.(*extensions.Ingress)
+	oldIngress := old.(*extensions.Ingress)
+	// status changes are not allowed to update spec
+	newIngress.Spec = oldIngress.Spec
+}
+
+func (ingressStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
```

Comment:
I think it's wise to validate the object as well, in addition to the update. That still assumes that validIngress + validUpdate = validIngress, but it's better than nothing (though in this context validating object doesn't fetch us much, it might in the future).